### PR TITLE
Add keymaps 

### PR DESCRIPTION
**Pull request by winterNebs**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ * |      |      |PrScr |      |      |             |      |      |      |      |      |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_LOWER] = LAYOUT( \
+  KC_GRV,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, \
+  KC_DEL,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    _______, \
+  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS, \
+  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, KC_PGUP, KC_UP,	 KC_PGDN, \
+  _______, _______, KC_PSCR, _______, _______, _______, _______, _______, _______, _______, _______, _______\
+),
```
Comment (drashna) [inline keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c:83]:
```suggestion
[_LOWER] = LAYOUT( 
  KC_GRV,  _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, 
  KC_DEL,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    _______, 
  KC_DEL,  KC_F1,   KC_F2,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_MINS, KC_EQL,  KC_LBRC, KC_RBRC, KC_BSLS, 
  _______, KC_F7,   KC_F8,   KC_F9,   KC_F10,  KC_F11,  KC_F12,  _______, _______, KC_PGUP, KC_UP,	 KC_PGDN, 
  _______, _______, KC_PSCR, _______, _______, _______, _______, _______, _______, _______, _______, _______
),
```

File: keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+ * | NKRO |      |      |      |      |             |      |      | Last | Vol- | Next |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_ADJUST] =  LAYOUT( \
+  RESET,   RGB_TOG, RGB_MOD, RGB_HUD, RGB_HUI, _______, _______, _______, _______, KC_WH_L, KC_WH_R, _______,\
+  RGB_SAD, RGB_SAI, RGB_VAD, RGB_VAI, AG_NORM, AG_SWAP, KC_HOME, KC_BTN1, KC_MS_U, KC_BTN2, KC_WH_U, _______, \
+  _______, _______, _______, QWERTY,  WORKMAN, _______, KC_END,  KC_MS_L, KC_MS_D, KC_MS_R, KC_WH_D, _______,\
+  _______, _______, _______, _______, _______, _______, KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, KC_VOLU, KC_MPLY, \
+  MAGIC_TOGGLE_NKRO, _______, _______, _______, _______, _______, _______, _______, _______, KC_MPRV, KC_VOLD, KC_MNXT \
+)
```
Comment (drashna) [inline keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c:104]:
```suggestion
[_ADJUST] =  LAYOUT( 
  RESET,   RGB_TOG, RGB_MOD, RGB_HUD, RGB_HUI, _______, _______, _______, _______, KC_WH_L, KC_WH_R, _______,
  RGB_SAD, RGB_SAI, RGB_VAD, RGB_VAI, AG_NORM, AG_SWAP, KC_HOME, KC_BTN1, KC_MS_U, KC_BTN2, KC_WH_U, _______, 
  _______, _______, _______, QWERTY,  WORKMAN, _______, KC_END,  KC_MS_L, KC_MS_D, KC_MS_R, KC_WH_D, _______,
  _______, _______, _______, _______, _______, _______, KC_LEFT, KC_DOWN, KC_UP,   KC_RGHT, KC_VOLU, KC_MPLY, 
  MAGIC_TOGGLE_NKRO, _______, _______, _______, _______, _______, _______, _______, _______, KC_MPRV, KC_VOLD, KC_MNXT 
)
```

File: keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+#ifdef AUDIO_ENABLE
+float tone_qwerty[][2]     = SONG(QWERTY_SOUND);
+float tone_dvorak[][2]     = SONG(DVORAK_SOUND);
+float tone_colemak[][2]    = SONG(COLEMAK_SOUND);
+#endif
+
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
```
Comment (drashna) [inline keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c:118]: These are depricated, and should be removed. 
```suggestion
```

File: keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        persistent_default_layer_set(1UL<<_QWERTY);
```
Comment (drashna) [inline keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c:124]: https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer
```suggestion
        set_single_persistent_default_layer(_QWERTY);
```

File: keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+      }
+      return false;
+      break;
+    case WORKMAN:
+      if (record->event.pressed) {
+        persistent_default_layer_set(1UL<<_WORKMAN);
```
Comment (drashna) [inline keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c:130]:
```suggestion
        set_single_persistent_default_layer(_WORKMAN);
```

File: keyboards/kyria/keymaps/winterNebs/keymap.c [hunk 6]
```diff
@@ -0,0 +1,743 @@
...
+		} 
+		else {
+			right = true;
+		}
+	}
+	#endif
+	switch (keycode) {
+		case QWERTY:
+			if (record->event.pressed) {
+				persistent_default_layer_set(1UL<<_QWERTY);
```
Comment (drashna) [inline keyboards/kyria/keymaps/winterNebs/keymap.c:222]:
```suggestion
				set_single_persistent_default_layer(_QWERTY);
```

File: keyboards/kyria/keymaps/winterNebs/keymap.c [hunk 7]
```diff
@@ -0,0 +1,743 @@
...
+			}
+			return false;
+			break;
+		case WORKMAN:
+			if (record->event.pressed) {
+				persistent_default_layer_set(1UL<<_WORKMAN);
```
Comment (drashna) [inline keyboards/kyria/keymaps/winterNebs/keymap.c:228]:
```suggestion
				set_single_persistent_default_layer(_WORKMAN);
```

File: keyboards/kyria/keymaps/winterNebs/rules.mk [hunk 8]
```diff
@@ -0,0 +1,15 @@
+OLED_DRIVER_ENABLE = yes   # Enables the use of OLED displays
+ENCODER_ENABLE = yes       # Enables the use of one or more 
+NKRO_ENABLE = yes
+WPM_ENABLE = yes
+EXTRAFLAGS += -flto
+CONSOLE_ENABLE = no     # Console for debug(+400)
+AUDIO_ENABLE   = no     # Audio output on port C6
+LINK_TIME_OPTIMIZATION_ENABLE = yes
```
Comment (drashna) [inline keyboards/kyria/keymaps/winterNebs/rules.mk:8]: You have the short form (now default/only form) below, as well.
```suggestion
```

File: keyboards/planck/keymaps/winterNebs/keymap.c [hunk 9]
```diff
@@ -0,0 +1,177 @@
...
+};
+
+
+//uint32_t layer_state_set_user(uint32_t state) {
+//  return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+//}
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
```
Comment (drashna) [inline keyboards/planck/keymaps/winterNebs/keymap.c:119]:
```suggestion
```

File: keyboards/planck/keymaps/winterNebs/keymap.c [hunk 10]
```diff
@@ -0,0 +1,177 @@
...
+
+bool held[7][6] = {{false}};
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  if(record->event.key.row != 3 && record->event.key.row != 7 ){
+    held[record->event.key.row][record->event.key.col] = record->event.pressed;
+  }
+  switch (keycode) {
+    case QWERTY:
+      if (record->event.pressed) {
+        persistent_default_layer_set(1UL<<_QWERTY);
```
Comment (drashna) [inline keyboards/planck/keymaps/winterNebs/keymap.c:130]:
```suggestion
        set_single_persistent_default_layer(_QWERTY);
```

File: keyboards/planck/keymaps/winterNebs/keymap.c [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+      }
+      return false;
+      break;
+    case WORKMAN:
+      if (record->event.pressed) {
+        persistent_default_layer_set(1UL<<_WORKMAN);
```
Comment (drashna) [inline keyboards/planck/keymaps/winterNebs/keymap.c:136]:
```suggestion
       set_single_persistent_default_layer(_WORKMAN);
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by winterNebs** [comment]
Thank you for taking the time to make these suggestions! 

A lot of these, I guess, were artifacts from the default keymaps of various boards, and I sort of just had the "if it ain't broke don't fix it" mentality.

File: keyboards/planck/keymaps/winterNebs/readme.md [hunk 12]
```diff
@@ -0,0 +1,2 @@
+# The Default Planck Layout
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/winterNebs/readme.md:1]:
```suggestion
# winterNebs' Planck keymap
```

File: keyboards/kyria/keymaps/winterNebs/rules.mk [hunk 13]
```diff
@@ -0,0 +1,13 @@
+OLED_DRIVER_ENABLE = yes   # Enables the use of OLED displays
+ENCODER_ENABLE = yes       # Enables the use of one or more 
+NKRO_ENABLE = yes
+WPM_ENABLE = yes
+EXTRAFLAGS += -flto
```
Comment (noroadsleft) [inline keyboards/kyria/keymaps/winterNebs/rules.mk:5]: `LTO_ENABLE = yes` supersedes this.

```suggestion
```

File: keyboards/kyria/keymaps/winterNebs/rules.mk [hunk 14]
```diff
@@ -0,0 +1,13 @@
...
+CONSOLE_ENABLE = no     # Console for debug(+400)
+AUDIO_ENABLE   = no     # Audio output on port C6
```
Comment (noroadsleft) [inline keyboards/kyria/keymaps/winterNebs/rules.mk:7]:
```suggestion
CONSOLE_ENABLE = no     # Console for debug
AUDIO_ENABLE   = no     # Audio output
```

File: keyboards/planck/keymaps/winterNebs/keymap.c [hunk 15]
```diff
@@ -0,0 +1,170 @@
...
+
+      layer_off(_LOWER);
+      return false;
+      break;
+	case HELP:
+	  if(record->event.pressed){
+        
+      }
+	  return false;
+      break;
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/winterNebs/keymap.c:155]: This keycode literally won't do anything.

```suggestion
```

File: keyboards/planck/keymaps/winterNebs/keymap.c [hunk 16]
```diff
@@ -0,0 +1,170 @@
...
+ * |      |      |      |QWERTY|WORKMAN|      |      | Help |      |      |      |      |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * |      |      |      |      |      |      |      |      | Home |  End | VOL+ | Play |
+ * |------+------+------+------+------+------+------+------+------+------+------+------|
+ * | NKRO |      |      |      |      |             |      |      | Prev | VOL- | Next |
+ * `-----------------------------------------------------------------------------------'
+ */
+[_ADJUST] = LAYOUT_planck_grid(
+    RESET,   DEBUG,   _______, _______, AU_ON,    AU_OFF, _______, _______, _______, _______, _______, _______ ,
+    _______, _______, _______, QWERTY, WORKMAN , _______, _______, HELP,	  _______, _______, _______, _______,
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/winterNebs/keymap.c:106]:
```suggestion
    _______, _______, _______, QWERTY, WORKMAN , _______, _______, _______, _______, _______, _______, _______,
```

File: keyboards/planck/keymaps/winterNebs/keymap.c [hunk 17]
```diff
@@ -0,0 +1,170 @@
...
+  _WORKMAN,
+  _LOWER,
+  _ADJUST,
+};
+
+enum planck_keycodes {
+  QWERTY = SAFE_RANGE,
+  WORKMAN,
+  LOWER,
+  HELP
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/winterNebs/keymap.c:32]:
```suggestion
```

File: keyboards/kyria/keymaps/winterNebs/keymap.c [hunk 18]
```diff
@@ -0,0 +1,743 @@
...
+#include QMK_KEYBOARD_H
+
+#ifdef CONSOLE_ENABLE
+	#include <print.h>
+#endif
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  WORKMAN,
+  LOWER,
+  RAISE,
```
Comment (noroadsleft) [inline keyboards/kyria/keymaps/winterNebs/keymap.c:25]: You don't need these here if they're just aliases for native keycodes (defined on Lines 38 and 39).

```suggestion
```

File: keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c [hunk 19]
```diff
@@ -0,0 +1,153 @@
...
+      return false;
+      break;
+    case ADJUST:
+      if (record->event.pressed) {
+        layer_on(_ADJUST);
+      } else {
+        layer_off(_ADJUST);
+      }
+      return false;
+      break;
```
Comment (noroadsleft) [inline keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c:150]: These are equivalent to just aliasing the native `MO()` keycodes.

You can do:

```suggestion
```

(1/3)

File: keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c [hunk 20]
```diff
@@ -0,0 +1,153 @@
...
+    _WORKMAN,
+    _LOWER,
+    _ADJUST,
+};
+
+enum custom_keycodes {
+  QWERTY = SAFE_RANGE,
+  WORKMAN,
+  LOWER,
+  ADJUST,
```
Comment (noroadsleft) [inline keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c:19]:
```suggestion
```

(2/3)

File: keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c [hunk 21]
```diff
@@ -0,0 +1,153 @@
...
+};
+
```
Comment (noroadsleft) [inline keyboards/keebio/nyquist/keymaps/winterNebs/keymap.c:21]:
```suggestion

#define LOWER MO(_LOWER)
#define ADJUST MO(_ADJUST)

```

(3/3)

**Review by noroadsleft — CHANGES_REQUESTED** [review]
GitHub tip: You can apply multiple suggestions to a single commit by using the [Files Changed tab](https://github.com/qmk/qmk_firmware/pull/10328/files).

**Review by winterNebs — COMMENTED** [review]
Ok looks good

File: keyboards/kyria/keymaps/winterNebs/keymap.c [hunk 22]
```diff
@@ -0,0 +1,741 @@
...
+		} else {
+			tap_code(KC_UP);
+		}
+	}
+}
+#endif
+void persistent_default_layer_set(uint16_t default_layer) {
+  eeconfig_update_default_layer(default_layer);
+  default_layer_set(default_layer);
+}
```
Comment (drashna) [inline keyboards/kyria/keymaps/winterNebs/keymap.c:194]: Could you remove this function?  

It's been deprecated in favor of the `set_single_persistent_default_layer` function (which is a core function).

https://docs.qmk.fm/#/ref_functions?id=setting-the-persistent-default-layer

File: keyboards/kyria/keymaps/winterNebs/keymap.c [hunk 23]
```diff
@@ -0,0 +1,741 @@
...
+	{{0, 4}, {1, 4}, {2, 4}, {3, 4}, {4, 4}, {5, 4}, {6, 4}, {7, 4}},
+	{{0, 5}, {1, 5}, {2, 5}, {3, 5}, {4, 5}, {5, 5}, {6, 5}, {7, 5}},
+	{{0, 6}, {1, 6}, {2, 6}, {3, 6}, {4, 6}, {5, 6}, {6, 6}, {7, 6}},
+	{{0, 7}, {1, 7}, {2, 7}, {3, 7}, {4, 7}, {5, 7}, {6, 7}, {7, 7}},
+	{{0, 0}, {1, 0}, {2, 0}, {3, 0}, {4, 0}, {5, 0}, {6, 0}, {7, 0}},
+	{{0, 1}, {1, 1}, {2, 1}, {3, 1}, {4, 1}, {5, 1}, {6, 1}, {7, 1}},
+	{{0, 2}, {1, 2}, {2, 2}, {3, 2}, {4, 2}, {5, 2}, {6, 2}, {7, 2}},
+	{{0, 3}, {1, 3}, {2, 3}, {3, 3}, {4, 3}, {5, 3}, {6, 3}, {7, 3}}
+};	
+
```
Comment (drashna) [inline keyboards/kyria/keymaps/winterNebs/keymap.c:266]: Honestly, I'd rather that this be pulled out of this PR and added to the keyboard, in a separate PR.

File: keyboards/planck/keymaps/winterNebs/keymap.c [hunk 24]
```diff
@@ -0,0 +1,163 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+
+extern keymap_config_t keymap_config;
```
Comment (drashna) [inline keyboards/planck/keymaps/winterNebs/keymap.c:19]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]
Also, could you make sure that the keymap folder names are all lowercase?

File: keyboards/kyria/keymaps/winterNebs/keymap.c (hunk 23 above)
Comment (winterNebs) [reply keyboards/kyria/keymaps/winterNebs/keymap.c:262]: I'll try and do that

Comment (winterNebs) [reply keyboards/kyria/keymaps/winterNebs/keymap.c]: Should be added in https://github.com/qmk/qmk_firmware/pull/10397 

I guess that should be merged first then I can remove the extra swap hands from my keymap?

Comment (noroadsleft) [reply keyboards/kyria/keymaps/winterNebs/keymap.c]: Should rebase this branch on `qmk:master` now.

Comment (winterNebs) [reply keyboards/kyria/keymaps/winterNebs/keymap.c]: done!

File: keyboards/keebio/nyquist/keymaps/winternebs/keymap.c [hunk 25]
```diff
@@ -0,0 +1,128 @@
+#include QMK_KEYBOARD_H
```
Comment (noroadsleft) [inline keyboards/keebio/nyquist/keymaps/winternebs/keymap.c:1]: Need a license header.

File: keyboards/planck/keymaps/winternebs/config.h [hunk 26]
```diff
@@ -0,0 +1,45 @@
+#pragma once
```
Comment (noroadsleft) [inline keyboards/planck/keymaps/winternebs/config.h:1]: Need a license header.

**Review by noroadsleft — CHANGES_REQUESTED** [review]

File: keyboards/kyria/keymaps/winternebs/rules.mk [hunk 27]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
```
Comment (noroadsleft) [inline keyboards/kyria/keymaps/winternebs/rules.mk:17]: This file doesn't need a license header.

```suggestion
```

File: keyboards/planck/keymaps/winternebs/config.h (hunk 26 above)
Comment (noroadsleft) [inline keyboards/planck/keymaps/winternebs/config.h:1]: This file *does* need a license header.

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]
Looks good to me.

**Review by drashna — APPROVED** [review]
LGTM, thanks!

**Pull request merged**